File: gyroTest/src/main/java/com/stulnikov/arduino/arduinoBt/gyrotest/gyrotest/dagger/ApplicationComponent.java
package com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.dagger;

import android.app.Application;

import com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.bluetooth.BluetoothController;
import com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.sensor.AccelerometerProvider;

import javax.inject.Singleton;

import dagger.Component;

/**
 * @author alexeystulnikov 2/28/17.
 */
@Component(modules = ApplicationModule.class)
@Singleton
public interface ApplicationComponent {
    Application getApplication();

    AccelerometerProvider getAccelerometer();

    BluetoothController getBluetooth();
}

File: gyroTest/src/main/kotlin/com/stulnikov/arduino/arduinoBt/gyrotest/gyrotest/dagger/ApplicationComponent.kt
package com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.dagger

import android.app.Application
import android.content.Context
import com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.bluetooth.BluetoothController
import dagger.Component
import javax.inject.Singleton

/**
 * @author alexeystulnikov 2/28/17.
 */
@Component(modules = [ApplicationModule::class])
@Singleton
interface ApplicationComponent {
    val context: Context
    val application: Application?
    val bluetooth: BluetoothController
}